core/server/api/v0.1/decorators/urls.js

```diff
@@ -0,0 +1,70 @@
+const urlService = require('../../../services/url');
+const {urlFor, makeAbsoluteUrls} = require('../../../services/url/utils');
+
+const urlsForPost = (post, options) => {
+    post.url = urlService.getUrlByResourceId(post.id);
+
+    if (options.columns && !options.columns.includes('url')) {
```

Comment:
Was thinking if this part shouldn't be rewritten as:
```javascript
if (options.columns && options.columns.includes('url')) {
    post.url = urlService.getUrlByResourceId(post.id);
}
```
As at the moment the the url attribute still stays there if just and empty `options` object is passed in (this seems like a bug)

Reply:
It's correct.

If you ask for `?columns=title`, you only want the title.

Reply:
You suggestion is a behaviour change. `columns` is not required

Reply:
@gargol

Reply:
This suggestion was more due to having to pass `columns` to the post url generation all the time see - https://github.com/TryGhost/Ghost/pull/9917/files#diff-c834d088d2557531b689ac7525a46744R23 . It seems a bit weird to pass empty array just to make sure **not** to get any extra columns back :thinking:

Reply:
Why do you have to pass it again? 🤔

Reply:
not important for now. just copying over previous behavior :)